File: C/guess.c
# include <stdio.h>
# include <stdlib.h>
# include <time.h>
# include <ctype.h>
#include <errno.h>

# define BUFFER 100

int readGuess();
void checkNumber();

int main() {
	int num, guess, max_size;
	char input[BUFFER];
	char *end;;

	// Set max size of random number
	max_size = 101;
	
	// Generate random num (not the optimal way...)
	srand(time(NULL));
	num = rand() % max_size;

	// Print welcome message
	printf("Hi! I'm thinking of a number between 0 and 100. ");
	printf("Can you guess it??\n");

	do {
		printf("\nPlease enter your guess: ");

		if (fgets(input, sizeof(input), stdin) != NULL) {

			// reset errno
			errno = 0;

			// convert string to int
			// If successful, check number.
			guess = strtol(input, NULL, 10);
			if (errno == EINVAL) {
				printf("Oops! That doesn't look valid!\n");
			} else {
				checkNumber(guess, num);
			}
		}
	} while (guess != num);

	return 0;
}

void checkNumber(int guess, int num) {

	if (guess < num) {
		printf("Higher than %d...\n", guess);
	} else if (guess > num) {
		printf("Lower than %d...\n", guess);
	} else if (guess == num) {
		printf("Yeah! It was %d!\n\n", num);
	}
}

File: Elixir/guess.ex
defmodule Guess do
  @moduledoc """
  A simple number guessing game.
  """

  @doc """
  Chooses secret number at random, then calls guess_and_check.
  Prints welcome/closing messages.
  """
  def play do
    secret_number = Enum.random(0..100)
    IO.puts "Welcome to a number guessing game in... elixir!"
    guess_count = guess_and_check(secret_number, 1)
    IO.puts "Yeah!! The secret number was #{secret_number}."
    IO.puts "You got it in #{guess_count} tries."
  end

  defp guess_and_check(secret_number, guess_count) do
    guess = get_guess()
    case guess do
      ^secret_number -> guess_count
      _ ->
        msg = if guess < secret_number, do: "higher!", else: "lower!"
        IO.puts "  You guessed #{guess}... Go #{msg}"
        guess_and_check(secret_number, guess_count+1)
    end
  end

  defp get_guess do
    guess = Integer.parse(IO.gets "Guess a number between 1 and 100: ")
    case guess do
      {n, _} when n in 1..100 -> n
      _ ->
        IO.puts "  Oops. That doesn't look like a valid guess..."
        get_guess()
    end
  end

end
